src/frontends/lean/decl_util.cpp

```diff
@@ -132,7 +132,7 @@ optional<name> heuristic_inst_name(name const & ns, expr const & type) {
     if (is_constant(arg_head)) {
         arg_name = const_name(arg_head);
     } else if (is_sort(arg_head) || is_sort_wo_universe(arg_head)) {
-        arg_name = "sort";
+        arg_name = sort_level(arg_head) == mk_level_zero() ? "Prop" : "sort";
```

Comment:
I think we can get `Type` working too:
```suggestion
        if (sort_level(arg_head) == mk_level_zero()) {
            arg_name = "Prop";
        } else if (optional<level> u1 = dec_level(u)) {
            arg_name = "Type";
        } else {
            arg_name = "sort";
        }
```
Can you add a test lean file that does something like:
```lean
class my_class (α : Sort*)

instance : my_class Prop := ⟨⟩
#check Prop.my_class

instance : my_class (Type*) := ⟨⟩
#check Type.my_class

instance : my_class (Sort*) := ⟨⟩
#check sort.my_class
```

Reply:
Or actually, just add to the existing tests here: https://github.com/leanprover-community/lean/blob/405273c925b7a088f09f78d640f5798a30f1f163/tests/lean/run/instance_naming.lean

Reply:
I'm not sure this will work; if you look at the only instance I could find of a `Type*/Type u` instance in `mathlib` ([here](https://github.com/leanprover-community/mathlib/blob/c2d042c6c21576685486c09827f6a7a66bd59a83/src/category_theory/preadditive/projective.lean#L99)) and try mess with it a little, you'll see the instance ends up being called `category_theory.enough_projectives`, instead of `sort.enough_projectives`.

Reply:
(Also, your code uses a single `=` in L137)

Reply:
I'll look into this a little bit tomorrow but don't currently understand the codebase enough to untangle why this doesn't work

Reply:
> (Also, your code uses a single = in L137)

That's deliberate :)

Reply:
Ohhh, I see! Should be `arg_head` anyways though

Reply:
Edited